Patch by Joshua Timberman <joshua@opscode.com>

Hi!

I hacked around with the githubredir.rb script and found what appears to be
a workable solution. Looks like github changed from the 'archives' URI to
'downloads', which is a bit more feature-rich in the webui, but for
whatever reason with the Merb project (and maybe others?), didn't get all
the tags.

Can you apply the attached diff for githubredir?

Thanks!

--- githubredir.rb      2010-03-22 21:35:51.000000000 -0600
+++ githubredir.rb.downloads    2010-03-22 21:40:37.000000000 -0600
@@ -58,15 +58,14 @@
 
         releases = {}
 
-        rels = doc / '#other_archives' / 'li'
+        rels = doc / '#downloads' / 'td' / 'a'
         raise RuntimeError, 'No releases found' if rels.empty?
 
-        rels.each do |li|
-          a = (li/'a')[0]
+        rels.each do |a|
           # In order for the links to end in .tar.gz (for uscan to be happy,
           # of course), we link back to ourselves with a tag - And the tag 
           # will just be sent over to github.
-          label = a.attributes['href'].gsub(/^.*archives\//, '')
+          label = a.attributes['href'].gsub(/^.*tarball\//, '') if a.inner_html =~ /tgz/
           link = '/github/%s/%s/%s.tar.gz' % [@author, @project, label]
           releases[label] = link
         end
@@ -104,7 +103,7 @@
 
   private
   def index_uri(branch='master')
-    'http://github.com/%s/%s/archives/%s' % [@author, @project, branch]
+    'http://github.com/%s/%s/downloads' % [@author, @project]
   end
 
   def html_for_tags(tags)
